projects/step-frontend/src/lib/os-plugins/modules/script-editor/components/script-editor/script-editor.component.html

```diff
@@ -1,14 +1,40 @@
 <mat-card>
   <mat-card-header>
-    <mat-card-title> Script Editor - Keyword: {{keyword?.attributes?.['name']}} </mat-card-title>
-    <button type="button" mat-icon-button (click)="execute()" matTooltip="Execute this plan" matTooltipPosition="below">
-      <step-icon name="play"></step-icon>
-    </button>
+    <mat-card-title> Script Editor - Keyword: {{keyword?.attributes?.['name']}}</mat-card-title>
+    <section>
+      <step-btn-group *ngIf="'kw-write' | hasRight | async" class="with-borders mr2">
+        <button
+          type="button"
+          mat-stroked-button
+          (click)="save()"
+          matTooltip="Save this plan"
+          matTooltipPosition="below"
+        >
+          <step-icon name="save"></step-icon>
+        </button>
+        <button
+          [ngClass]="{ 'color-green': noChanges && isAfterSave, 'color-black': !noChanges }"
+          disabled
+          type="button"
+          mat-stroked-button
+        >
+          {{ noChanges ? (isAfterSave ? 'Saved' : 'No changes') : isAfterSave ? 'Saved' : 'Pending changes' }}
```

Comment:
I really like the ? operator, but chaining multiple together makes the code hard to read. I think the easiest solution would be to add line breaks like this:

```suggestion
 {{
    noChanges
      ? (isAfterSave ? 'Saved' : 'No changes')
      : (isAfterSave ? 'Saved' : 'Pending changes')
  }}
```

Reply:
Done